File: Day 9/0 Two Sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;

        for (int i = 0; i < nums.size(); i++) {
            int requiredNum = target - nums[i];

            if (map.count(requiredNum)) {
                return {map[requiredNum], i};
            }
            
            if (!map.count(nums[i])) {
                map[nums[i]] = i;
            }
        }
        return {};
    }
};

File: Day 9/0 Two Sum/solution.js
var twoSum = function(nums, target) {
    var map = new Map();
    for (var i = 0; i < nums.length; i++) {
        var requiredNum = target - nums[i];
        if (map.has(requiredNum)) {
            return [map.get(requiredNum), i];
        }
        if (!map.has(nums[i])) {
            map.set(nums[i], i);
        }
    }
    return [];
};
